dart/math/Helpers.h

```diff
@@ -68,13 +65,22 @@ inline int delta(int _i, int _j) {
   return 0;
 }
 
-inline int sgn(double _a) {
-  if (_a < 0)
-    return -1;
-  else if (_a == 0)
-    return 0;
-  else
-    return 1;
+template <typename T> inline constexpr
+int sgn(T x, std::false_type)
```

Comment:
I think my personal preference would be to name the function ``sign``, because it only adds one letter and makes the purpose of the function completely obvious (whereas ``sgn`` *could* be an acronym for something else).

That being said, I don't object to using ``sgn``.